### ISSUE TITLE
loss_uniform is negative

### ISSUE THREAD

**Issue by hkzhang-git** [facebookresearch/spreadingvectors#3]
Inspired by your work, I attempted to reduce feature dimension via source code released here. 

I carried out dimension reduction experiments with compression factor of 4 on two datasets, including GIST1M (960 d, compressed to 240 d) and Deep1M (256 d, compressed to 64 d. PCA is not used to compressed this dataset to 96 d). Both datasets are download from https://www.cse.cuhk.edu.hk/systems/hash/gqr/datasets.html. 

Experimental results are strange
1) during training, the second loss item loss_uniform is negative. 
2) I also attempted to reproduce experiments in your Table1. On Deep1M,  Recall 1@5 of Catalyst+IVFPQ (nlist=8, M=16, nbits=8) is 47.9%, which is 9.3 percentage points higher that that of IVFPQ(nlist=8, M=16, nbits=8). But, on GIST 1M, Recall 1@5 of Catalyst+IVFPQ(nlist=8, M=60, nbits=8) is 37.4%, 7.8 pencentage points lower than that of IVFPQ(nlist=8, M=60, nbits=8). It seems that Catalyst does not work on GIST1M.

**Comment by alexandresablayrolles** [comment]
1. The loss can be negative. Although it derives from an entropy estimator, it lacks a few terms to make it a proper estimator so there are not reasons why it should be positive.
2. Table 1 in the paper is not using IVFPQ but rather a lattice encoding or an OPQ encoding (without IVF). However regarding your experiments, I would encourage you to cross-validate the hyper-parameters for GIST1M, as the parameters by default are tuned for Deep1M/BIgANN1M and a lattice quantizer.

**Comment by hkzhang-git** [comment]
Thanks for your reply. It has solved all my concerns. I will follow your suggestion to look for proper hyper-parameters for GIST1M. Thanks again.
